js/crm.drupal7.js

```diff
@@ -0,0 +1,35 @@
+// https://civicrm.org/licensing
+(function($) {
+  "use strict";
+
+  // Buffer for 2 events - we're waiting for both the Drupal menu and the Civi menu to be rendered before proceeding
+  var menuReadyEvents = {
+    adminMenu: $.Deferred(),
+    crmMenu: $.Deferred()
+  };
+
+  $.when(menuReadyEvents.adminMenu, menuReadyEvents.crmMenu)
+    .done(hideMenuToggleButtonForNonAdminUsers);
+  $(document)
+    .on('crmLoad', '#civicrm-menu', menuReadyEvents.crmMenu.resolve)
+    .ready(menuReadyEvents.adminMenu.resolve);
+
+  /**
+   * Hides the Menu Toggle Button when the Admin Menu is not available for the user.
+   * It also positions the CiviCRM Menu in the right position in case it was displayed
+   * under where the Admin Menu would have been. This avoids displaying an empty gap.
+   */
+  function hideMenuToggleButtonForNonAdminUsers() {
+    if (!$('#toolbar').length) {
+      $('#crm-menubar-toggle-position').hide();
+
+      if (CRM.menubar.position === 'below-cms-menu') {
+        CRM.menubar.togglePosition();
+      }
+
+      // When the menubar gets re-rendered, this will prevent the toggle button from being added
+      CRM.menubar.initializePosition = function(){};
```

Comment:
Better use 

```js
CRM.menubar.initializePosition = CRM._.noop;
```

If we check the reference for `CRM.menubar.initializePosition` and find it's assigned to a random empty function, we might be scratching our heads as to what happened, but `_.noop` means that this was done on purpose.

Even better would be to lock this functionality:

```
CRM.menubar.lockMenuPosition();
```

If the menubar is locked, `initializePosition()` should throw a warning. There should also be a `isMenuPositionLocked()` function so we can safely run `initializePosition()` without triggering a warning. This is a more complicated approach, but it explains better what happened, why we should not use this function. 

When Triggering a function, if nothing happens might lead to confusion so it's best to be explicit about it through a warning message (either a `throw` or a `console.error`, etc.). Also, another file should not override the original `CRM.menubar`. 

Just leaving this as a comment, not an actual action item.